R/tar_quarto_files.R

```diff
@@ -60,12 +60,31 @@ tar_quarto_files <- function(path = ".", profile = NULL) {
 tar_quarto_files_document <- function(path) {
   info <- quarto::quarto_inspect(input = path)
   out <- list(sources = path)
+
+  # Collect data about output files.
   for (format in info$formats) {
     out$output <- c(
       out$output,
       file.path(dirname(path), format$pandoc$`output-file`)
     )
   }
+
+  # Collect data about input files.
+  for (myfile in names(info$fileInformation)) {
+    out$input <- c(
+      out$input,
+      # `myfile` are relative paths starting from `path`.
+      myfile,
+      # `includeMap` contains relative paths starting from `myfile`.
+      file.path(
+        dirname(path),
+        info$fileInformation[[myfile]]$includeMap$target
+      )
+    )
+  }
+  # Input `path` is also part of `input`, thus remove it again.
+  out$input <- out$input[out$input != path]
```

Comment:
Could we do `out$input <- unique(out$input)` instead? That way we'll still have it if `quarto_inspect()` drops it in a future version of `quarto`.

Reply:
Sure! Sorry, I understood the documentation that a file can only be either in `input`, `output`, or `sources`. But if this is not the case, it's much easier to keep it.

`unique(out$input)` is not necessary, as `unique` will be called anyway in [line 54](https://github.com/ropensci/tarchetypes/pull/199/files/d6388e3b094576c16bc321bd5f1964ee00e89d01#diff-524d2e5ff1b127cc08c5332702a792937f52dbde7e19afc212e54004f591f88eR54).

Reply:
I didn't notice before the failing tests that this impacts also other tests in `quarto_files`. But I think it is worth it because it makes dependency detection more comprehensive.